### PR TITLE
enh(api): user and global parameters endpoints

### PR DESCRIPTION
**Pull request by adr-mo**
This PR intends to add 2 endpoints
- GET /configuration/users/current/parameters - retrieves information regarding the current user
- GET /administration/parameters - retrieves global informations

**Example of results**

```
For global information
{
    "monitoring_default_refresh_interval": "15",
    "monitoring_default_downtime_duration": "3600"
}

For user information
{
    "name": "centreon centreon",
    "alias": "admin",
    "email": "centreon@localhost",
    "timezone": "Europe/Paris",
    "locale": "en_US"
}
```

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 19.10.x
- [ ] 20.04.x
- [x] 20.10.x
- [x] 21.04.x (master)

<h2> How this pull request can be tested ? </h2>

Can be tested by using those endpoints and check the consistency of the values returned.

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: doc/API/centreon-api-v2.yaml [hunk 1]
```diff
@@ -269,6 +269,42 @@ paths:
...
+                  email:
+                    type: string
+                    description: Email of the current user
+                    example: admin@localhost
+                  timezone:
+                    type: string
+                    description: Timezone for the current user
+                    example: Europe/Paris
+                  locale:
+                    type: string
```
Comment (kduret) [inline doc/API/centreon-api-v2.yaml:303]:
```suggestion
                    type: string
                    nullable: true
```

File: src/Centreon/Application/Controller/Administration/ParametersController.php [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+    /**
+     * Needed to make response "more readable"
+     */
+    private const KEY_NAME_CONCORDANCE = [
+        self::DEFAULT_REFRESH_INTERVAL => 'monitoring_default_refresh_interval',
+        self::DEFAULT_DOWNTIME_DURATION => 'monitoring_default_downtime_duration'
+    ];
+
+    public function __construct(OptionServiceInterface $optionService)
```
Comment (kduret) [inline src/Centreon/Application/Controller/Administration/ParametersController.php:53]: missing phpdoc

File: src/Centreon/Domain/Option/OptionService.php [hunk 3]
```diff
@@ -50,7 +50,11 @@ public function __construct(OptionRepositoryInterface $repository)
      */
     public function findSelectedOptions(array $optionsToFind): array
     {
-        $optionsFound = $this->repository->findAllOptions();
+        try {
+            $optionsFound = $this->repository->findAllOptions();
+        } catch (\Throwable $ex) {
+            throw new \Exception(_('Error when retrieving selected options'));
```
Comment (kduret) [inline src/Centreon/Domain/Option/OptionService.php:56]:
```suggestion
            throw new \Exception(_('Error when retrieving selected options'), 0, $ex);
```

**Review by loiclau — DISMISSED** [review]

**Review by jeremyjaouen — DISMISSED** [review]

File: src/Centreon/Application/Controller/UserController.php [hunk 4]
```diff
@@ -60,6 +60,27 @@ public function getActionsAuthorization(): View
...
+
+        $userParameters = [
+            'name' => $user->getName(),
+            'alias' => $user->getAlias(),
+            'email' => $user->getEmail(),
+            'timezone' => $user->getTimezone()->getName(),
+            'locale' => $user->getLocale()
+        ];
+
+        return $this->view($userParameters);
```
Comment (callapa) [inline src/Centreon/Application/Controller/UserController.php:82]:
```suggestion
        return $this->view([
            'name' => $user->getName(),
            'alias' => $user->getAlias(),
            'email' => $user->getEmail(),
            'timezone' => $user->getTimezone()->getName(),
            'locale' => $user->getLocale()
        ]);
```

Comment (adr-mo) [reply src/Centreon/Application/Controller/UserController.php]: Yep that's way better thanks :100:

**Review by jeremyjaouen — DISMISSED** [review]

File: src/Centreon/Application/Controller/Administration/ParametersController.php [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+     * Entry point to get global parameters stored in options table
+     *
+     * @return View
+     */
+    public function getParameters(): View
+    {
+        $this->denyAccessUnlessGrantedForApiConfiguration();
+
+        $parameters = [];
+        $options = [];
```
Comment (kduret) [inline src/Centreon/Application/Controller/Administration/ParametersController.php:73]: Not need to initialized this value cause it's reassigned after
```suggestion
```

File: tests/api/Context/UserParametersContext.php [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
```
Comment (kduret) [inline tests/api/Context/UserParametersContext.php:21]:
```suggestion

declare(strict_types=1);

```

**Review by kduret — APPROVED** [review]

**Review by jeremyjaouen — APPROVED** [review]

**Pull request merged**